pennylane_lightning/core/src/simulators/lightning_qubit/gates/tests/Test_OpToMemberFuncPtr.cpp

```diff
@@ -38,25 +38,28 @@ template <typename EnumClass> constexpr auto allGateOps() {
         std::make_integer_sequence<uint32_t,
                                    static_cast<uint32_t>(EnumClass::END)>{}));
 }
-template <class PrecisionT, class ParamT, class GateImplemenation,
+template <class PrecisionT, class ParamT, class GateImplementation,
           uint32_t gate_idx>
 constexpr bool testAllGatesImplementedIter() {
     if constexpr (gate_idx <
-                  static_cast<uint32_t>(Pennylane::Gates::GateOperation::END)) {
+                  static_cast<uint32_t>(Pennylane::Gates::GateOperation::END) -
+                      static_cast<uint32_t>(
+                          Pennylane::Gates::GateOperation::BEGIN)) {
         constexpr auto gate_op = static_cast<GateOperation>(gate_idx);
         static_cast<void>(
-            GateOpToMemberFuncPtr<PrecisionT, ParamT, GateImplemenation,
+            GateOpToMemberFuncPtr<PrecisionT, ParamT, GateImplementation,
                                   gate_op>::value);
         return testAllGatesImplementedIter<PrecisionT, ParamT,
-                                           GateImplemenation, gate_idx + 1>();
+                                           GateImplementation, gate_idx + 1>();
     } else {
         return true;
     }
 }
-template <class PrecisionT, class ParamT, class GateImplemenation>
-constexpr bool testAllGatesImplemeted() {
-    return testAllGatesImplementedIter<PrecisionT, ParamT, GateImplemenation,
-                                       0>();
+template <class PrecisionT, class ParamT, class GateImplementation>
+constexpr bool testAllGatesImplemented() {
+    return testAllGatesImplementedIter<PrecisionT, ParamT, GateImplementation,
+                                       static_cast<uint32_t>(
+                                           GateOperation::BEGIN)>();
```

Comment:
Wait, so these numbers were just raw passed in previously, rather than being referenced from their origin? Good catch on tracking these things.

Reply:
> Wait, so these numbers were just raw passed in previously, rather than being referenced from their origin? 

Yep :)